File: data/recomposed_projects/llama-3-70b-instruct/body/commons-graph/evosuite-test/src/test/org/apache/commons/graph/coloring/DefaultColoringAlgorithmsSelector_ESTest.py
from __future__ import annotations
import time
import re
import mock
import os
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.graph.Mapper import *
from src.main.org.apache.commons.graph.UndirectedGraph import *
from src.main.org.apache.commons.graph.coloring.ColoredVertices import *
from src.main.org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector import *

# from src.test.org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector_ESTest_scaffolding import *
from src.main.org.apache.commons.graph.model.MutableSpanningTree import *
from src.main.org.apache.commons.graph.model.UndirectedMutableGraph import *
from src.main.org.apache.commons.graph.weight.Monoid import *

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
# from src.main.org.evosuite.runtime.ViolatedAssumptionAnswer import *
# from src.main.org.evosuite.shaded.org.mockito.Mockito import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Answer import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Stubber import *


class DefaultColoringAlgorithmsSelector_ESTest(unittest.TestCase):

    def test14(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        linkedHashSet0 = set()
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )
        coloredVertices0 = (
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()
        )
        self.assertEqual(0, coloredVertices0.getRequiredColors())

    def test13(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        linkedHashSet0 = set()
        integer0 = -1862
        linkedHashSet0.add(integer0)
        undirectedMutableGraph0.addVertex(integer0)
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )
        coloredVertices0 = (
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()
        )
        coloredVertices1 = (
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(
                coloredVertices0
            )
        )
        self.assertEqual(1, coloredVertices1.getRequiredColors())

    def test12(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        linkedHashSet0 = set()
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )
        coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingGreedyAlgorithm()
        self.assertEqual(0, coloredVertices0.getRequiredColors())

    def test11(self) -> None:

        undirected_mutable_graph0 = UndirectedMutableGraph()
        linked_hash_set0 = set()
        integer0 = -1862
        undirected_mutable_graph0.add_vertex(integer0)
        default_coloring_algorithms_selector0 = DefaultColoringAlgorithmsSelector(
            undirected_mutable_graph0, linked_hash_set0
        )

        try:
            default_coloring_algorithms_selector0.applying_greedy_algorithm()
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            self.verify_exception(
                "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e
            )

    def test10(self) -> None:

        undirected_mutable_graph0 = UndirectedMutableGraph()
        linked_hash_set0 = set()
        integer0 = -1862
        linked_hash_set0.add(None)
        undirected_mutable_graph0.add_vertex(integer0)
        default_coloring_algorithms_selector0 = DefaultColoringAlgorithmsSelector(
            undirected_mutable_graph0, linked_hash_set0
        )
        colored_vertices0 = (
            default_coloring_algorithms_selector0.applying_back_tracking_algorithm0()
        )
        self.assertEqual(1, colored_vertices0.get_required_colors())

    def test09(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        integer0 = Integer(340)
        undirectedMutableGraph0.addVertex(integer0)
        linkedHashSet0 = LinkedHashSet()
        integer1 = Integer(1)
        undirectedMutableGraph0.addVertex(integer1)
        coloredVertices0 = ColoredVertices()
        undirectedMutableGraph0.addEdge(integer0, coloredVertices0, integer1)
        linkedHashSet0.add(integer1)
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )

        try:
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            verifyException(
                "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e
            )

    def test08(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        integer0 = Integer(-212)
        undirectedMutableGraph0.addVertex(integer0)
        linkedHashSet0 = LinkedHashSet()
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )
        coloredVertices0 = ColoredVertices()

        with pytest.raises(RuntimeError):
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(
                coloredVertices0
            )
            verifyException(
                "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector",
                RuntimeError,
            )

    def test07(self) -> None:

        linkedHashSet0 = set()
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            None, linkedHashSet0
        )

        with pytest.raises(RuntimeError):
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()
            self.fail("Expecting exception: RuntimeError")

        verifyException(
            "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e
        )

    def test06(self) -> None:

        monoid0 = unittest.mock.Mock(spec=Monoid)
        monoid0.identity.return_value = None

        mapper0 = unittest.mock.Mock(spec=Mapper)

        mutableSpanningTree0 = MutableSpanningTree(monoid0, mapper0)

        linkedHashSet0 = set()
        mutableSpanningTree0.addVertex(linkedHashSet0)

        comparable0 = unittest.mock.Mock(spec=Comparable)
        comparable0.toString.return_value = None
        linkedHashSet0.add(comparable0)

        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            mutableSpanningTree0, linkedHashSet0
        )

        with pytest.raises(RuntimeError):
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()

        assert "org.apache.commons.graph.model.BaseGraph" in str(e.value)

    def test05(self) -> None:

        undirected_mutable_graph0 = UndirectedMutableGraph()
        linked_hash_set0 = set()
        default_coloring_algorithms_selector0 = DefaultColoringAlgorithmsSelector(
            undirected_mutable_graph0, linked_hash_set0
        )

        with pytest.raises(RuntimeError):
            default_coloring_algorithms_selector0.applyingBackTrackingAlgorithm1(None)
            assert False, "Expecting exception: RuntimeError"

    def test04(self) -> None:

        linkedHashSet0 = set()
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            None, linkedHashSet0
        )
        coloredVertices0 = ColoredVertices()

        with pytest.raises(RuntimeError):
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(
                coloredVertices0
            )
            self.fail("Expecting exception: RuntimeError")

        self.verifyException(
            "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e
        )

    def test02(self) -> None:

        linkedHashSet0 = set()
        monoid0 = Mock(Monoid)
        monoid0.identity.return_value = None
        mapper0 = Mock(Mapper)
        mutableSpanningTree0 = MutableSpanningTree(monoid0, mapper0)
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            mutableSpanningTree0, linkedHashSet0
        )
        coloredVertices0 = ColoredVertices()
        coloredVertices1 = (
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(
                coloredVertices0
            )
        )
        self.assertIs(coloredVertices1, coloredVertices0)

    def test01(self) -> None:

        undirected_mutable_graph0 = UndirectedMutableGraph()
        integer0 = Integer(-212)
        undirected_mutable_graph0.add_vertex(integer0)
        linked_hash_set0 = LinkedHashSet()
        integer1 = Integer(-1246)
        linked_hash_set0.add(integer1)
        integer2 = Integer(-3954)
        undirected_mutable_graph0.add_vertex(integer2)
        default_coloring_algorithms_selector0 = DefaultColoringAlgorithmsSelector(
            undirected_mutable_graph0, linked_hash_set0
        )
        colored_vertices0 = (
            default_coloring_algorithms_selector0.applying_greedy_algorithm()
        )
        self.assertEqual(1, colored_vertices0.get_required_colors())

    def test00(self) -> None:

        undirectedMutableGraph0 = UndirectedMutableGraph()
        linkedHashSet0 = set()
        integer0 = Integer(2238)
        undirectedMutableGraph0.addVertex(integer0)
        coloredVertices0 = ColoredVertices()
        integer1 = Integer(2238)
        undirectedMutableGraph0.addEdge(integer0, coloredVertices0, integer1)
        linkedHashSet0.add(integer0)
        defaultColoringAlgorithmsSelector0 = DefaultColoringAlgorithmsSelector(
            undirectedMutableGraph0, linkedHashSet0
        )

        try:
            defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0()
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            verifyException(
                "org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e
            )

File: java_projects/cleaned_final_projects_evosuite/commons-graph/src/test/java/org/apache/commons/graph/coloring/DefaultColoringAlgorithmsSelector_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.coloring;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.LinkedHashSet;
import org.apache.commons.graph.Mapper;
import org.apache.commons.graph.UndirectedGraph;
import org.apache.commons.graph.coloring.ColoredVertices;
import org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector;
import org.apache.commons.graph.model.MutableSpanningTree;
import org.apache.commons.graph.model.UndirectedMutableGraph;
import org.apache.commons.graph.weight.Monoid;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class DefaultColoringAlgorithmsSelector_ESTest extends DefaultColoringAlgorithmsSelector_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer0 = new Integer(2238);
      undirectedMutableGraph0.addVertex(integer0);
      ColoredVertices<Integer, Integer> coloredVertices0 = new ColoredVertices<Integer, Integer>();
      Integer integer1 = new Integer(2238);
      undirectedMutableGraph0.addEdge(integer0, coloredVertices0, integer1);
      linkedHashSet0.add(integer0);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      Integer integer0 = new Integer((-212));
      undirectedMutableGraph0.addVertex(integer0);
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer1 = new Integer((-1246));
      linkedHashSet0.add(integer1);
      Integer integer2 = new Integer((-3954));
      undirectedMutableGraph0.addVertex(integer2);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingGreedyAlgorithm();
      assertEquals(1, coloredVertices0.getRequiredColors());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Monoid<ColoredVertices<Integer, Integer>> monoid0 = (Monoid<ColoredVertices<Integer, Integer>>) mock(Monoid.class, new ViolatedAssumptionAnswer());
      doReturn((Object) null).when(monoid0).identity();
      Mapper<Integer, ColoredVertices<Integer, Integer>> mapper0 = (Mapper<Integer, ColoredVertices<Integer, Integer>>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      MutableSpanningTree<Integer, Integer, ColoredVertices<Integer, Integer>> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, ColoredVertices<Integer, Integer>>(monoid0, mapper0);
      DefaultColoringAlgorithmsSelector<Integer, Integer, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, Integer, Integer>(mutableSpanningTree0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = new ColoredVertices<Integer, Integer>();
      ColoredVertices<Integer, Integer> coloredVertices1 = defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(coloredVertices0);
      assertSame(coloredVertices1, coloredVertices0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, Integer, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, Integer, Integer>((UndirectedGraph<Integer, Integer>) null, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingGreedyAlgorithm();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      LinkedHashSet<ColoredVertices<Integer, Integer>> linkedHashSet0 = new LinkedHashSet<ColoredVertices<Integer, Integer>>();
      DefaultColoringAlgorithmsSelector<Integer, Integer, ColoredVertices<Integer, Integer>> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, Integer, ColoredVertices<Integer, Integer>>((UndirectedGraph<Integer, Integer>) null, linkedHashSet0);
      ColoredVertices<Integer, ColoredVertices<Integer, Integer>> coloredVertices0 = new ColoredVertices<Integer, ColoredVertices<Integer, Integer>>();
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(coloredVertices0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1((ColoredVertices<Integer, Integer>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // PartialColoredVertex must be not null
         //
         verifyException("org.apache.commons.graph.utils.Assertions", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Monoid<Object> monoid0 = (Monoid<Object>) mock(Monoid.class, new ViolatedAssumptionAnswer());
      doReturn((Object) null).when(monoid0).identity();
      Mapper<Integer, Object> mapper0 = (Mapper<Integer, Object>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      MutableSpanningTree<Object, Integer, Object> mutableSpanningTree0 = new MutableSpanningTree<Object, Integer, Object>(monoid0, mapper0);
      LinkedHashSet<Comparable<Object>> linkedHashSet0 = new LinkedHashSet<Comparable<Object>>();
      mutableSpanningTree0.addVertex(linkedHashSet0);
      Comparable<Object> comparable0 = (Comparable<Object>) mock(Comparable.class, new ViolatedAssumptionAnswer());
      doReturn((String) null).when(comparable0).toString();
      linkedHashSet0.add(comparable0);
      DefaultColoringAlgorithmsSelector<Object, Integer, Comparable<Object>> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Object, Integer, Comparable<Object>>(mutableSpanningTree0, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, Integer, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, Integer, Integer>((UndirectedGraph<Integer, Integer>) null, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      Integer integer0 = new Integer((-212));
      undirectedMutableGraph0.addVertex(integer0);
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = new ColoredVertices<Integer, Integer>();
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(coloredVertices0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      Integer integer0 = new Integer(340);
      undirectedMutableGraph0.addVertex(integer0);
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer1 = new Integer(1);
      undirectedMutableGraph0.addVertex(integer1);
      ColoredVertices<Integer, Integer> coloredVertices0 = new ColoredVertices<Integer, Integer>();
      undirectedMutableGraph0.addEdge(integer0, coloredVertices0, integer1);
      linkedHashSet0.add(integer1);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer0 = new Integer((-1862));
      linkedHashSet0.add((Integer) null);
      undirectedMutableGraph0.addVertex(integer0);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
      assertEquals(1, coloredVertices0.getRequiredColors());
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer0 = new Integer((-1862));
      undirectedMutableGraph0.addVertex(integer0);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      // Undeclared exception!
      try { 
        defaultColoringAlgorithmsSelector0.applyingGreedyAlgorithm();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.coloring.DefaultColoringAlgorithmsSelector", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingGreedyAlgorithm();
      assertEquals(0, coloredVertices0.getRequiredColors());
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      Integer integer0 = new Integer((-1862));
      linkedHashSet0.add(integer0);
      undirectedMutableGraph0.addVertex(integer0);
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
      ColoredVertices<Integer, Integer> coloredVertices1 = defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm1(coloredVertices0);
      assertEquals(1, coloredVertices1.getRequiredColors());
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>> undirectedMutableGraph0 = new UndirectedMutableGraph<Integer, ColoredVertices<Integer, Integer>>();
      LinkedHashSet<Integer> linkedHashSet0 = new LinkedHashSet<Integer>();
      DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer> defaultColoringAlgorithmsSelector0 = new DefaultColoringAlgorithmsSelector<Integer, ColoredVertices<Integer, Integer>, Integer>(undirectedMutableGraph0, linkedHashSet0);
      ColoredVertices<Integer, Integer> coloredVertices0 = defaultColoringAlgorithmsSelector0.applyingBackTrackingAlgorithm0();
      assertEquals(0, coloredVertices0.getRequiredColors());
  }
}
